backend/packages/Upgrade/src/api/services/MoocletExperimentService.ts

```diff
@@ -231,24 +264,549 @@ export class MoocletExperimentService extends ExperimentService {
+   *    - Throws an error if ineligible changes are detected for non-INACTIVE experiments
+   * 4. Updates the Upgrade experiment first using the parent class implementation
+   * 5. Updates Mooclet resources based on detected changes:
+   *    - For INACTIVE or ENROLLING states: Updates policy parameters
+   *    - For INACTIVE state only, if design changes exist:
+   *      a. Updates outcome variable name if changed
+   *      b. Removes conditions that were deleted
+   *      c. Modifies conditions that were changed
+   *      d. Adds new conditions that were created
+   * 6. Returns the updated experiment if all operations succeed
+   * 7. If any step fails, rolls back all changes made to Mooclet resources
+   *
+   * @param manager - EntityManager instance for database transactions
+   * @param params - Object containing experiment data, current user, and logger
+   * @returns Promise resolving to the updated ExperimentDTO
+   * @throws Error if ineligible changes are detected or if any operation fails
+   */
+
+  private async handleEditMoocletTransaction(manager: EntityManager, params: SyncEditParams): Promise<ExperimentDTO> {
+    const { experimentDTO: incomingExperiment, currentUser, logger } = params;
+    const rollbackRef: EditRollbackRef = {
+      revertPolicyParameters: null,
+      revertOutcomeVariableName: null,
+      restoreVersions: null,
+      revertVersionModifications: null,
+      removeVersions: null,
+      currentMoocletExperimentRef: null,
+      currentExperiment: null,
+    };
+    let currentMoocletExperimentRef: MoocletExperimentRef;
+
+    try {
+      // 1. detect the changes we might care about
+      currentMoocletExperimentRef = await this.getMoocletExperimentRefByUpgradeExperimentId(incomingExperiment.id);
+
+      const currentPolicyParametersResponse = await this.moocletDataService.getPolicyParameters(
+        currentMoocletExperimentRef.policyParametersId,
+        logger
+      );
+
+      const currentOutcomeVariableResponse = await this.moocletDataService.getVariable(
+        currentMoocletExperimentRef.variableId,
+        logger
+      );
+
+      const allowedInactiveStateChanges = await this.detectExperimentDesignChanges(
+        incomingExperiment,
+        currentMoocletExperimentRef,
+        currentOutcomeVariableResponse.name
+      );
+
+      if (incomingExperiment.state !== EXPERIMENT_STATE.INACTIVE && allowedInactiveStateChanges) {
+        const error = {
+          message: '[Mooclet Edit] Ineligible changes detected for an active Mooclet experiment.',
+          changes: allowedInactiveStateChanges,
+        };
+        logger.error({ error: JSON.stringify(error) });
+        throw new Error(JSON.stringify(error));
+      }
+
+      logger.debug({
+        message: '[Mooclet Edit] Experiment changes affect Mooclet resources, begin sync.',
+        changes: allowedInactiveStateChanges,
+      });
+
+      const currentExperiment = await this.experimentRepository.findOne({
```

Comment:
This can be fetched in parallel. (Not that important if it affects code readability)